searx/engines/discourse.py

```diff
@@ -0,0 +1,96 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+# lint: pylint
+"""Discourse Forums
+
+"""
+
+from urllib.parse import urlencode
+import html
+
+from dateutil import parser
+
+from searx.exceptions import SearxEngineException
+
+about = {
+    "website": "https://discourse.org/",
+    "wikidata_id": "Q15054354",
+    "official_api_documentation": "https://docs.discourse.org/",
+    "use_official_api": True,
+    "require_api_key": False,
+    "results": "JSON",
+}
+
+base_url = None
+search_endpoint = '/search.json?'
+
+# ['latest', 'likes', 'views', 'latest_topic']
+api_order = 'likes'
+paging = True
+
+
+def request(query, params):
+    """https://docs.discourse.org/#tag/Search/operation/search"""
+    if len(query) <= 2:
+        raise SearxEngineException(
+            'For engines using discourse the length of the search query must be at least 3 chars!'
+        )
+
+    args = urlencode(
+        {
+            'q': query + f' order:{api_order}',
+            'page': params['pageno'],
+        }
+    )
+
+    params['url'] = base_url + search_endpoint + args
+    params['headers'] = {
+        'Accept': 'application/json, text/javascript, */*; q=0.01',
+        'X-Requested-With': 'XMLHttpRequest',
+    }
+
+    return params
+
+
+def response(resp):
+
+    results = []
+    json_data = resp.json()
+
+    if ('topics' or 'posts') not in json_data.keys():
+        return []
+
+    for post, result in zip(json_data['posts'], json_data['topics']):
```

Comment:
python's `zip` function won't work here .. does not what we expect:

- `topics` and `posts` list do not have the same number of items
- the sorting of both lists is not the same / in the example below, `posts[4]` refers to topic id found in `topics[1]`

.. I will fix it in my review / give me some time, its in work ..

![image](https://github.com/searxng/searxng/assets/554536/4828faf6-700c-4ac5-81f1-cb609141987c)

Reply:
Thats right. Good catch.